source/Octostache.Tests/ParserFixture.cs

```diff
@@ -39,5 +41,44 @@ public void AListOfVariablesCanBeExtracted()
             result.Should().NotContain("item");
             result.Should().NotContain("item.Address");
         }
+
+        [Fact]
+        public void ParsedTemplateIsConvertibleBackToString()
+        {
+            // This test verifies that it is possible to convert a parsed template ToString
+            // And then find individual tokens within the template by calling ToString on each one individually
+            
+            var template = @"
+                #{var}
+                #{filter | Format #{option}}
+                #{nestedParent[#{nestedChild}]}
+                #{if ifvar}#{if ifnested}#{else}debug#{/if}#{/if}
+                #{if comparison != ""value"" }true#{/if}
+                #{each item in List}
+                   #{item.Address}
+                #{/each}
+                ##{ignored}
+                #{MyVar | Match ""a b""}
+                #{MyVar | StartsWith Ab}
+                #{MyVar | Match ""ab[0-9]+""}
+                #{MyVar | Match #{pattern}}
+                #{MyVar | Contains "" b(""}
+                #{ | NowDateUtc}
+                #{MyVar | UriPart IsFile}
+            ";
+
+            TemplateParser.TryParseTemplate(template, out var parsedTemplate, out string error).Should().BeTrue();
```

Comment:
My only comment here would be if we can split out the whole "if evaluation is true, the `parsedTemplate` should not be nulll and `error` should be null" into its own test, then make this test all about just the template check. Then add an extra comment, even a 1-liner, just to better explain why its doing this the replace on each value to end up with empty.

Ultimately nothing I'd consider a blocker, so feel free to merge before or after any tidy up.